src/main/js/isd.js

```diff
@@ -59,22 +59,90 @@
 
         };
 
-        /* process regions */
+        /* Filter body contents - Only process what we need within the offset and discard regions not applicable to the content */
+        var body = {};
+        var activeRegions = {};
+
+        /* gather any regions that might have showBackground="always" and show a background */
+        initialShowBackground = tt.head.styling.initials[imscStyles.byName.showBackground.qname];
+        initialbackgroundColor = tt.head.styling.initials[imscStyles.byName.backgroundColor.qname];
+        initialbackgroundImage = tt.head.styling.initials[imscStyles.byName.backgroundImage.qname];
+        for (var layout_child in tt.head.layout.regions)
+        {
+            if (tt.head.layout.regions.hasOwnProperty(layout_child)) {
+                region = tt.head.layout.regions[layout_child];
+                showBackground = region.styleAttrs[imscStyles.byName.showBackground.qname] || initialShowBackground;
+                backgroundColor = region.styleAttrs[imscStyles.byName.backgroundColor.qname] || initialbackgroundColor;
```

Comment:
```suggestion
```

Does TTML2 say that the absence of backgroundColor means a region is not active?

Reply:
No, but that's not what this code is checking for. 

This loop is:
a) making a copy list of all the region ids and 
b) ensuring that any regions that _might_ display a background are included in the set we iterate through later. 

The filter could be made more precise, but there's limited benefit in repeating the checks that will happen later anyway. The idea is only to avoid excluding regions that have `showBackground` set to `always` and that set some `backgroundColor`.  It might turn out that the background color is completely transparent, or the opacity is 0%: any regions like that will be processed, and the presentation consequences generated when the later loop calls `isdProcessContentElement()` for them.